### PR TITLE
Rewrite with typescript.

### PR DESCRIPTION
**Pull request by yooonspace**
Rewrite with typescript.
Please publish the module to version 3.0.0.

### CODE REVIEW

File: .babelrc [hunk 1]
```diff
@@ -1,8 +1,11 @@
...
-  }
+  "presets": [
+    "@babel/preset-env",
+    "@babel/preset-typescript",
+    "@babel/preset-react"
+  ],
+  "plugins": [
+    "@babel/plugin-proposal-class-properties",
+    "@babel/plugin-proposal-object-rest-spread"
+  ]
```
Comment (hustcc) [inline .babelrc]: 使用了 typescript，是否可以丢弃 babel build？

File: .eslintrc [hunk 2]
```diff
@@ -9,8 +9,6 @@
   },
   "parser": "babel-eslint",
   "rules": {
-    "quotes": [2, "single"],
```
Comment (hustcc) [inline .eslintrc]: 代码中还是使用单引号吧 ^_^

File: index.d.ts [hunk 3]
```diff
@@ -1,13 +1,13 @@
...
+  style?: object;
 }
 
-declare module 'timeago-react' {
-    export default class TimeAgo extends React.Component<Props> { }
-}
\ No newline at end of file
+declare module "timeago-react" {
+  export default class TimeAgo extends React.Component<Props> {}
+}
```
Comment (hustcc) [inline index.d.ts]: 使用 ts 之后，自动生成 d.ts，删除掉手工编写的类型定义文件？

File: lib/timeago-react.js [hunk 4]
```diff
@@ -1,140 +1,63 @@
-'use strict';
```
Comment (hustcc) [inline lib/timeago-react.js]: lib 目录放到 gitignore 吧

File: src/timeago-react.tsx [hunk 5]
```diff
@@ -0,0 +1,45 @@
+import React, { RefObject, TimeHTMLAttributes, useEffect, useRef } from "react";
```
Comment (hustcc) [inline src/timeago-react.tsx]: 使用 hook 之后是不是意味着 react 版本号只支持 16.8 之后版本吧，感觉可能不是很合适~

Comment (yooonspace) [reply src/timeago-react.tsx]: I think that babel support react hooks for past versions. I'll check it

File: lib/timeago-react.js (hunk 4 above)
Comment (yooonspace) [reply lib/timeago-react.js]: ok

**Comment by yooonspace** [comment]
I will accept your request and PR again.

**Comment by hustcc** [comment]
> I will accept your request and PR again.

Sorry, I mistook you are a Chinese I knew. I will rewrite with english.

File: tests/test.spec.js [hunk 6]
```diff
@@ -1,6 +1,10 @@
 import React from 'react';
```
Comment (hustcc) [inline tests/test.spec.js]: I think you can have a try with [jest-electron](https://github.com/hustcc/jest-electron), I think will be better then mock the component.

It's not necessary for this pr.

**Comment by yooonspace** [comment]
I'll open the PR again once the edit is complete.

**Pull request closed without merging**